### ISSUE TITLE
[BUG] master - CacheHash fails for viewfiles because of path resolving

### ISSUE THREAD

**Issue by GDmac** [bolt/bolt#4948]
#4947 changes backend assets inclusion

```php
// Twig: {{ cachehash(paths.app ~ 'view/css/lib.css') }}
// resolves to: cachehash('/boltdir/app/view/css/lib.css')

// Inside Twig htmlhandler:: cachehash
$fullPath = $this->app['resources']->getPath('root') . '/' . $fileName;

// But getPath(root) will actually try to get rootpath because
// first checks if (array_key_exists($name . 'path', $this->paths)
// paths.root == '/boltdir/'
// getPath(root) == '/Users/gdmac/sites/boltdir/'
// ---------------------------------------------------------------------------
// Same for paths.app vs asking getPath(app)

// Anyway
// ---------------------------------------------------------------------------
$fullpath == '/Users/gdmac/sites/boltdir/' /* rootpath */ . '/' . '/boltdir/app/view/css/lib.css';
// i.e. double boltdir (and double slash)
is_readable($fullpath) == false; // !!!
return;
```

**Comment by GawainLynch** [comment]
@CarsonF and I found this earlier today and we have a branch almost ready to go… result soon… watch this space.

**Comment by GDmac** [comment]
Added bonus issue (in my opinion)

getPath(root) in config/resourcemanager/getPathObject
actually fetches "rootpath" instead of "root" (intentional?)
because first checks (array_key_exists($name . 'path', $this->paths)
will never return app.paths[root], always return app.paths[rootpath]

**Comment by GDmac** [comment]
Bonus moved to #4949

**Comment by GawainLynch** [comment]
Fixed in #4959
